evalml/preprocessing/data_splitters/training_validation_split.py

```diff
@@ -43,4 +43,4 @@ def split(self, X, y=None):
                 list: indices to split data into training and test set
         """
         train, test = train_test_split(np.arange(X.shape[0]), test_size=self.test_size, train_size=self.train_size, shuffle=self.shuffle, stratify=self.stratify, random_state=self.random_state)
-        return [(train, test)]
+        return iter([(train, test)])
```

Comment:
I believe this was necessary because `BaseCrossValidator` subclasses are expected to return iterators. I had to update the tests to call `list(...)` 🤷‍♂️